concordium-consensus/src/Concordium/GlobalState/Persistent/LMDB.hs

```diff
@@ -688,3 +693,44 @@ writeFinalizationComposite finRec blocktss = resizeOnFull (finRecSize + blocksSi
           digestSize + 64 + digestSize + (32 * Prelude.length vs) + 48 + 64
     blocksSize = sum . map (\b -> 2*digestSize + fromIntegral (LBS.length . fst . fst $ b)) $ serializedBlocksTss
     tssSize = (* (2 * digestSize + 16)) . sum . map (Prelude.length . snd) $ blocktss
+
+-- If an explicitly finalized block does not satisfy a predicate, delete it, together with its
+-- finalization record, all blocks implicitly finalized by its finalization records and all
+-- associated transaction statuses. Returns the last finalized block hash available in the unrolled
+-- treestate/blockstate on success.
+unrollTreeStateWhile :: (MonadLogger m, MonadIO m, IsProtocolVersion pv, FixedSizeSerialization st) =>
```

Comment:
We have to be somewhat careful how we use the lmdb. We use a reader-writer lock to make sure that there is only the one active transactions when the db is being resized (see https://github.com/Concordium/concordium-node/pull/182 )

The way this function is currently used I think it's all good, since we are still in the setup stage. But please document the assumption here that nobody else is using the database while this function is working.